### DISCUSSION TITLE
Why anyhow::Result instead of std::result::Result?

### DISCUSSION THREAD

**Discussion by coder0xff** [fzyzcjy/flutter_rust_bridge#1070] [Q&A] [answered]
I'm guessing `anyhow::Result` does something really important that `std::result::Result` doesn't do. What is it?

**Comment by fzyzcjy** [comment, marked as answer]
> does something really important

No. I choose it _initially_ because of the convenience that anyhow brings ;)

However, std Result has no problems and can be supported. Here is a PR about it: https://github.com/fzyzcjy/flutter_rust_bridge/pull/582

**Reply by coder0xff** [reply]
Rust is new to me. I'd appreciate your sharing more of your insights on the advantages.

**Reply by fzyzcjy** [reply]
Maybe have a look at the anyhow doc itself, and google why people use it in (normal) rust projects. That said, I totally agree that std Result is more general and thus powerful. The #582 is almost done (but a bit out of date), so if you are interested, feel free to continue on the PR!
